src/python/Utils/MemoryCache.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+"""
+Simple in-memory and non-thread safe cache
+"""
+
+from __future__ import (print_function, division)
+from time import time
+
+
+class MemoryCacheException(Exception):
+    def __init__(self, message):
+        super(MemoryCacheException, self).__init__(message)
+
+
+class MemoryCache(object):
+
+    __slots__ = ["lastUpdate", "expiration", "_cache"]
+
+    def __init__(self, expiration, initialData=None):
+        """
+        Initializes cache object
+
+        :param expiration: expiration time in seconds
+        :param initialData: initial value for the cache
+        """
+        self.lastUpdate = int(time())
+        self.expiration = expiration
+        self._cache = initialData
+
+    def __contains__(self, item):
+        """
+        Check whether item is in the current cache
+        :param item: a simple object (string, integer, etc)
+        :return: True if the object can be found in the cache, False otherwise
+        """
+        return item in self._cache
+
+    def isCacheExpired(self):
+        """
+        Evaluate whether the cache has already expired, returning
+        True if it did, otherwise it returns False
+        """
+        return self.lastUpdate + self.expiration < int(time())
+
+    def getCache(self):
+        """
+        Raises an exception if the cache has expired, otherwise returns
+        its data
+        """
+        if self.isCacheExpired():
+            expiredSince = int(time()) - (self.lastUpdate + self.expiration)
+            raise MemoryCacheException("Memory cache expired for %d seconds" % expiredSince)
+        return self._cache
+
+    def setCache(self, inputData):
+        """
+        Refresh the cache with the content provided (refresh its expiration as well)
+        :param inputData: data to store in the cache
+        """
+        self.lastUpdate = int(time())
+        self._cache = inputData
+
+    def addItemToCache(self, inputItem):
+        """
+        Adds new item(s) to the cache, without resetting its expiration.
+        NOTE that it will only work for list, set and dictionary types.
+        :param inputItem: additional item to be added to the current cached data
+        """
+        if isinstance(self._cache, set) and isinstance(inputItem, (list, set)):
+            # extend another list or set into a set
+            self._cache.update(inputItem)
+        elif isinstance(self._cache, set):
```

Comment:
That's a good comment Todor, I didn't want to add a check there because them we limit the object types to be stored in this cache (like, we can no longer store complex objects)